extra_addons/account_check_deposit/models/account_deposit.py

```diff
@@ -275,3 +276,31 @@ def delete_check_payment(self):
         self.ensure_one()
         self.write({'check_deposit_id': False})
         return True
+
+    @api.onchange('partner_id')
+    def onchange_partner_id(self):
+        if self.partner_id:
```

Comment:
Do not update the value if the check_holder_name has already been manually changed before partner change.